base/harmful/harmful.c

```diff
@@ -65,7 +65,6 @@ TRAP(gethostbyaddr)
 TRAP(gethostbyname)
 TRAP(gethostbyname2)
 TRAP(gethostent)
-TRAP(getlogin)
```

Comment:
Where do we use it? It's not thread safe and can cause races with other functions:

```
ATTRIBUTES         [top](https://man7.org/linux/man-pages/man3/getlogin.3.html#top_of_page)

       For an explanation of the terms used in this section, see
       [attributes(7)](https://man7.org/linux/man-pages/man7/attributes.7.html).
       ┌──────────────┬───────────────┬────────────────────────────────┐
       │ Interface    │ Attribute     │ Value                          │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ getlogin()   │ Thread safety │ MT-Unsafe race:getlogin        │
       │              │               │ race:utent sig:ALRM timer      │
       │              │               │ locale                         │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ getlogin_r() │ Thread safety │ MT-Unsafe race:utent sig:ALRM  │
       │              │               │ timer locale                   │
       ├──────────────┼───────────────┼────────────────────────────────┤
       │ cuserid()    │ Thread safety │ MT-Unsafe race:cuserid/!string │
       │              │               │ locale                         │
       └──────────────┴───────────────┴────────────────────────────────┘

       In the above table, utent in race:utent signifies that if any of
       the functions [setutent(3)](https://man7.org/linux/man-pages/man3/setutent.3.html), [getutent(3)](https://man7.org/linux/man-pages/man3/getutent.3.html), or [endutent(3)](https://man7.org/linux/man-pages/man3/endutent.3.html) are used
       in parallel in different threads of a program, then data races
       could occur.  getlogin() and getlogin_r() call those functions,
       so we use race:utent to remind users.
```